airflow/models/dag.py

```diff
@@ -3318,12 +3317,12 @@ def bulk_write_to_db(
             curr_outlet_references = curr_orm_dag and curr_orm_dag.task_outlet_dataset_references
             for task in dag.tasks:
                 dataset_outlets: list[Dataset] = []
-                dataset_alias_outlets: list[DatasetAlias] = []
+                dataset_alias_outlets: set[DatasetAlias] = set()
                 for outlet in task.outlets:
                     if isinstance(outlet, Dataset):
                         dataset_outlets.append(outlet)
                     elif isinstance(outlet, DatasetAlias):
-                        dataset_alias_outlets.append(outlet)
+                        dataset_alias_outlets.add(outlet)
```

Comment:
Makes me wonder why the datasets are in a list instead. Maybe out of scope for this PR though.

Reply:
According to https://github.com/apache/airflow/blob/a7b0dc25c0201e3e69726aa6033102128d8d9046/airflow/models/dag.py#L3300, it seems we want to preserve the order.

Reply:
I feel it might not be required for dataset alias (or maybe dataset even?)

I would like to know whether @dstandish has some thoughts on this. Thanks!

Reply:
As this comment is not critical and the PR fixes a serious issue, I will merge it after CI passes, but we can continue the discussion here.